packages/medusa/src/api/routes/admin/inventory-items/list-inventory-items.ts

```diff
@@ -0,0 +1,219 @@
+ *     style: form
+ *     explode: false
+ *     description: Locations ids to search for.
+ *     schema:
+ *       type: array
+ *       items:
+ *         type: string
+ *   - (query) id {string} id to search for.
+ *   - (query) sku {string} sku to search for.
+ *   - (query) origin_country {string} origin_country to search for.
+ *   - (query) mid_code {string} mid_code to search for.
+ *   - (query) material {string} material to search for.
+ *   - (query) hs_code {string} hs_code to search for.
+ *   - (query) weight {string} weight to search for.
+ *   - (query) length {string} length to search for.
+ *   - (query) height {string} height to search for.
+ *   - (query) width {string} width to search for.
+ *   - (query) requires_shipping {string} requires_shipping to search for.
+ * x-codeSamples:
+ *   - lang: JavaScript
+ *     label: JS Client
+ *     source: |
+ *       import Medusa from "@medusajs/medusa-js"
+ *       const medusa = new Medusa({ baseUrl: MEDUSA_BACKEND_URL, maxRetries: 3 })
+ *       // must be previously logged in or use api token
+ *       medusa.admin.inventoryItems.list()
+ *       .then(({ inventory_items }) => {
+ *         console.log(inventory_items.length);
+ *       });
+ *   - lang: Shell
+ *     label: cURL
+ *     source: |
+ *       curl --location --request GET 'https://medusa-url.com/admin/inventory-items' \
+ *       --header 'Authorization: Bearer {api_token}' \
+ *       --header 'Content-Type: application/json'
+ * security:
+ *   - api_token: []
+ *   - cookie_auth: []
+ * tags:
+ *   - Inventory Items
+ * responses:
+ *   200:
+ *     description: OK
+ *     content:
+ *       application/json:
+ *         schema:
+ *           $ref: "#/components/schemas/AdminInventoryItemsListWithVariantsAndLocationLevelsRes"
+ *   "400":
+ *     $ref: "#/components/responses/400_error"
+ *   "401":
+ *     $ref: "#/components/responses/unauthorized"
+ *   "404":
+ *     $ref: "#/components/responses/not_found_error"
+ *   "409":
+ *     $ref: "#/components/responses/invalid_state_error"
+ *   "422":
+ *     $ref: "#/components/responses/invalid_request_error"
+ *   "500":
+ *     $ref: "#/components/responses/500_error"
+ */
+
+type ResponseInventoryItemWithVariantsAndLocationLevels =
+  Partial<InventoryItemDTO> & {
+    location_levels?: InventoryLevelDTO[]
+    variants?: ProductVariant[]
+  }
+
+export default async (req: Request, res: Response) => {
+  const inventoryService: IInventoryService =
+    req.scope.resolve("inventoryService")
+  const productVariantInventoryService: ProductVariantInventoryService =
+    req.scope.resolve("productVariantInventoryService")
+  const productVariantService: ProductVariantService = req.scope.resolve(
+    "productVariantService"
+  )
+
+  const { filterableFields, listConfig } = req
+  const { skip, take } = listConfig
+
+  let locationIds: string[] = []
+
+  if (filterableFields.location_id) {
+    locationIds = Array.isArray(filterableFields.location_id)
```

Comment:
suggestion(non blocking): Why don't we change to location_ids and always expect an array? It can also be am array of one location that way it is always consistent and less pron to error

Reply:
> also, as an overall, I havn't seen any tests as we do usually for the end points handlers

all tests related to Inventory and Stock Location modules will come in a next iteration.